Core/src/main/java/com/plotsquared/core/generator/HybridGen.java

```diff
@@ -217,34 +222,170 @@ public void generateChunk(@NonNull ScopedQueueCoordinator result, @NonNull PlotA
-                                    true
-                            );
+                            placeSchem(hybridPlotWorld, result, relativeX[x], relativeZ[z], x, z, true, false);
                         }
                     } else {
                         // plot
                         for (int y = startY; y < hybridPlotWorld.PLOT_HEIGHT; y++) {
                             result.setBlock(x, y, z, hybridPlotWorld.MAIN_BLOCK.toPattern());
                         }
-                        result.setBlock(x, hybridPlotWorld.PLOT_HEIGHT, z,
-                                hybridPlotWorld.TOP_BLOCK.toPattern()
-                        );
+                        result.setBlock(x, hybridPlotWorld.PLOT_HEIGHT, z, hybridPlotWorld.TOP_BLOCK.toPattern());
                         if (hybridPlotWorld.PLOT_SCHEMATIC) {
-                            placeSchem(hybridPlotWorld, result, relativeX[x], relativeZ[z], x, z,
-                                    false
-                            );
+                            placeSchem(hybridPlotWorld, result, relativeX[x], relativeZ[z], x, z, false, false);
                         }
                     }
                 }
             }
         }
     }
 
+    @Override
+    public boolean populateChunk(final ScopedQueueCoordinator result, final PlotArea settings) {
+        HybridPlotWorld hybridPlotWorld = (HybridPlotWorld) settings;
+        if (!hybridPlotWorld.populationNeeded()) {
+            return false;
+        }
+        // Coords
+        Location min = result.getMin();
+        int bx = min.getX() - hybridPlotWorld.ROAD_OFFSET_X;
+        int bz = min.getZ() - hybridPlotWorld.ROAD_OFFSET_Z;
+        // The relative X-coordinate (within the plot) of the minimum X coordinate
+        // contained in the scoped queue
+        short relativeOffsetX;
+        if (bx < 0) {
+            relativeOffsetX = (short) (hybridPlotWorld.SIZE + (bx % hybridPlotWorld.SIZE));
+        } else {
+            relativeOffsetX = (short) (bx % hybridPlotWorld.SIZE);
+        }
+        // The relative Z-coordinate (within the plot) of the minimum Z coordinate
+        // contained in the scoped queue
+        short relativeOffsetZ;
+        if (bz < 0) {
+            relativeOffsetZ = (short) (hybridPlotWorld.SIZE + (bz % hybridPlotWorld.SIZE));
+        } else {
+            relativeOffsetZ = (short) (bz % hybridPlotWorld.SIZE);
+        }
+        boolean allRoad = true;
+        boolean overlap = false;
+
+        // The X-coordinate of a given X coordinate, relative to the
+        // plot (Counting from the corner with the least positive
+        // coordinates)
+        short[] relativeX = new short[16];
+        boolean[] insideRoadX = new boolean[16];
+        boolean[] insideWallX = new boolean[16];
+        short offsetX = relativeOffsetX;
+        while (offsetX >= hybridPlotWorld.SIZE) {
+            offsetX -= hybridPlotWorld.SIZE;
```

Comment:
See comment below about modulo.
Why is overlap not set here?

Reply:
Overlap should have only been set in the for loop for the chunk itself, not the while loop outside